src/goto-instrument/code_contracts.cpp

```diff
@@ -1595,9 +1622,23 @@ exprt assigns_clauset::compatible_expression(
     {
       if(first_iter)
       {
-        current_target_compatible =
-          target->compatible_expression(*called_target);
-        first_iter = false;
+        // TODO: generalize this. Currently only supports pointers
+        // See GitHub issue #6087 for further details
+        if(target->target_type == assigns_clause_targett::target_type::Scalar)
+        {
+          current_target_compatible = or_exprt(
+            equal_exprt(
+              called_target->get_direct_pointer(),
+              null_pointer_exprt(
+                to_pointer_type(called_target->get_direct_pointer().type()))),
```

Comment:
Same comment as before about checking validity, not just `NULL`-ness.

But, why do we have this code duplication?
cc: @feliperodri

Reply:
This is for the case where we are using the `--enforce-contract` flag. So for any assignment, asside fromchecking if the assigned variable is part of the assigns clause, I am also asking CBMC to consider the case that the assigned variable may be NULL/invalid.